### ISSUE TITLE
Binary UUID : PR Wanted? 

### ISSUE THREAD

**Issue by bradennapier** [sequelize/sequelize#10265]
#9272

I wrote out a skeleton for how we can support UUID Binary type as recommended by MySQL team and many other articles to store uuid as binary(16).  

I’m happy to look into implementing it as a PR for official support as a data type if it would be welcomed by the team as an addition.  

My time is somewhat limited in coming months so don’t want to do unless it’s a desired feature :) 

Specifically take a look at the last post where I implement the uuid stuff client side.

**Comment by sushantdhiman** [comment]
Yes

**Comment by bradennapier** [comment]
@sushantdhiman So I put that PR together but realized there are probably quite a few places that I would be missing potentially important pieces.  I extracted the core values into two packages and published them to npm and tested.

---

Specifically around how to handle the new `BINARY` type (I played with using `UUIDV1.BINARY` as the default value for this which seems to work then having a `BINARY` type as the module below defines.

I wasn't sure how to best handle the `virtual` field in this case in a way that works similar to precedent.  It's fairly important IMO to be able to still get the human-readable values.  Seems some kind of transform hook would be good to always provide to the user as a string but always save to db as the binary.  This may already be available that I may be missing.

---

https://github.com/odo-network/binary-uuid

https://github.com/odo-network/sequelize-binary-uuid

```javascript
import Sequelize from "sequelize";
import { withBinaryUUID } from "sequelize-binary-uuid";

const User = sequelize.define(
  "User",
  withBinaryUUID(
    {
      // any other fields here
      someKey: Sequelize.DataTypes.TEXT
    },
    {
      primaryID: "id", // default
      virtualID: "uuid", // default
      field: {
        primaryKey: true
      }
    }
  )
);
```
